snapcraft/internal/meta.py

```diff
@@ -160,6 +161,12 @@ def _record_manifest_and_source_snapcraft_yaml(self):
                 yaml.dump(annotated_snapcraft, manifest_file)
 
     def _annotate_snapcraft(self, data):
+        image_info = os.environ.get('SNAPCRAFT_IMAGE_INFO')
+        if image_info:
+            image_info_dict = json.loads(image_info)
```

Comment:
I'm thinking maybe this shouldn't be picking what fields to include, since the invoking snapcraft may be newer than the one it's calling in the container. So if in the future we want to add or drop fields, it'd be good if the invoking snapcraft was in control.
Also, some error handling would probably be good? Log a warning like `Couldn't parse JSON image info`. Again, thinking to rather be conservative because of the invoking/ container snapcraft relationship.

Reply:
Ok, I'm raising an error if the parsing doesn't work. And I'm checking if the keys are present on the dict before recording them in the manifest.
I also removed the filtering on the host.

Reply:
Actually I meant to suggest the reverse. I'll try to explain more clearly. The `lxd.py` code path is the invoking snapcraft, which can be newer than the snapcraft being called inside the container. So if you choose the keys in `lxd.py` we're more flexible in the future.

Reply:
I get it now, and I think you are right. Leave the logic to the host, and on the guest just throw whatever is in the variable into the file.

Reply:
Yeah. Thanks for changing it!